packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -0,0 +1,46 @@
+interface AnalyticsEvent {
+    name: string;
+    properties: {
+        [key: string]: string;
+    };
+}
+
+let eventQueue: AnalyticsEvent[] = [];
+
+const setEvent = (event: AnalyticsEvent): void => {
+    eventQueue.push(event);
+    localStorage.setItem('pending_events', JSON.stringify(eventQueue));
+};
+
+const trackEventWithCache = (event: AnalyticsEvent): void => {
+    if (window.rudderanalytics) {
+        window.rudderanalytics.track(event.name, event.properties);
+    } else {
+        setEvent(event);
+        handleCachedEvents();
```

Comment:
```suggestion
```
this should be moved to the if statement instead of else,
as soon as we have the analytics ready it should send them.